File: benchmarks/bench_6.go
package main

import (
	"fmt"
	"time"
)

func string_concat() int64 {
    s := ""
    i := int64(0)
    for i < 2000 {
        s += "a"
        i += 1
    }
    return int64(len(s))
}

func main() {
    t1 := time.Now()
    res := string_concat()
    t2 := time.Now()
    fmt.Println(res)
    fmt.Printf("%.6f\n", t2.Sub(t1).Seconds())
}

File: benchmarks/bench_6.py
import time
import sys
sys.setrecursionlimit(200000)

def string_concat():
    s = ""
    i = 0
    while i < 2000:
        s += "a"
        i += 1
    return len(s)


t1 = time.perf_counter()
res = string_concat()
t2 = time.perf_counter()
print(res)
print(f"{t2 - t1:.6f}")
